pytensor/tensor/type.py

```diff
@@ -101,7 +101,7 @@ def __init__(
         if str(dtype) == "floatX":
             self.dtype = config.floatX
         else:
-            if np.obj2sctype(dtype) is None:
+            if np.dtype(dtype).type is None:
```

Comment:
This needs a `try/except` now? `np.dtype` does not return `None` for invalid types, it raises

Reply:
This was changed in a later commit I assume? No biggie